menuinst/win32.py

```diff
@@ -251,26 +254,28 @@ def remove(self):
 
     def create(self, remove=False):
         args = []
+        # cmd is our root installation interpreter
+        cmd = join(unicode_prefix, u"pythonw.exe").replace("\\", "/")
+        # each of these roll the subprocess cmd into args
         if "pywscript" in self.shortcut:
-            cmd = join(self.menu.prefix, u"pythonw.exe").replace("\\", "/")
+            subprocess_cmd = join(self.menu.prefix, u"pythonw.exe").replace("\\", "/")
             args = self.shortcut["pywscript"].split()
-            args = get_python_args_for_subprocess(self.menu.prefix, args, cmd)
+            args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
         elif "pyscript" in self.shortcut:
-            cmd = join(self.menu.prefix, u"python.exe").replace("\\", "/")
+            subprocess_cmd = join(self.menu.prefix, u"python.exe").replace("\\", "/")
             args = self.shortcut["pyscript"].split()
-            args = get_python_args_for_subprocess(self.menu.prefix, args, cmd)
+            args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
         elif "webbrowser" in self.shortcut:
-            cmd = join(unicode_prefix, u'pythonw.exe')
+            subprocess_cmd = join(unicode_prefix, u'pythonw.exe')
             args = ['-m', 'webbrowser', '-t', self.shortcut['webbrowser']]
         elif "script" in self.shortcut:
-            cmd = self.shortcut["script"].replace('/', '\\')
+            subprocess_cmd = join(unicode_prefix, u"pythonw.exe").replace("\\", "/")
+            args.append(self.shortcut["script"].replace('/', '\\'))
             extend_script_args(args, self.shortcut)
-            args = get_python_args_for_subprocess(self.menu.prefix, args, cmd)
-            cmd = join(unicode_prefix, u"pythonw.exe").replace("\\", "/")
+            args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
         elif "system" in self.shortcut:
-            cmd = substitute_env_variables(
-                     self.shortcut["system"],
-                     self.menu.dir).replace('/', '\\')
+            args.append(substitute_env_variables(self.shortcut["system"],
```

Comment:
This won't work since the above `cmd` will be used for the `create_shortcut`'s `path` argument, hence the shortcut will be something like `pythonw.exe system_command args...`. Resolution:
1. restore previous behavior, or
2. or make this also use `python cwp env_prefix system_command args` instead of `system_command args`

Without knowing the implications I'd be in favor of the second option since why should the `"system"` case not use the `cwp`-adjusted environment?